tests/test_rerere.py

```diff
@@ -365,9 +340,13 @@ def flip_last_two_commits(repo: Repository, ed: Editor) -> None:
         )
 
 
-def dedent(text: str) -> bytes:
-    return textwrap.dedent(text).encode()
+def dedent_as_bytes(text: str) -> bytes:
+    return dedent(text).encode()
+
+
+def hunks(diff: bytes) -> str:
+    return diff[diff.index(b"@@") :].decode()
 
 
-def hunks(diff: bytes) -> bytes:
-    return diff[diff.index(b"@@") :]
+def leftover_index(diff: bytes) -> str:
```

Comment:
I'd drop the `leftover_index()` function, it barely does anything

Reply:
True, the function itself is hardly useful, but what it's there for is to save two lines at each call site compared to naming it by variable:

    leftover_index = ….decode()
    assert leftover_index == ""

Reply:
then pass it only the repo argument and make it compute `repo.git("diff", "-U0", "HEAD")`?

Reply:
makes sense. I might do that. The reason I didn't was that the verbosity wasn't in the way of readability.

Reply:
done

Reply:
actually the "index" part of the function name is not great because it also includes unstaged changes.
An easy fix would be to use `diff-index` instead of `diff`

Reply:
Hm, let's rename it. How about `uncommitted_changes`?

There used to be leftover staged changes, but now, the test is just using this to check that there are no changes introduced, staged or unstaged.